### PR TITLE
Actually implementing dbGetInfo for R package

### LINKED ISSUE #1600: R API support for `DBI::dbGetInfo()`
I am not sure if this is intended behavior so please feel free to close this issue if irrelevant.

The output of a `duckdb::duckdb()` DBI connection in `DBI::dbGetInfo()` is useless and I wonder if it is intended or not. I'll try to show by comparing to the output of an `RSQLite::SQLite()` connection. 

I am depending on this function to get the path to the db. Is there a better method to get the path to the db via the connection?

Thanks!

``` r
con_sqlite <- DBI::dbConnect(RSQLite::SQLite(), tempfile())
con_duckdb <- DBI::dbConnect(duckdb::duckdb(), tempfile())
DBI::dbGetInfo(con_sqlite)
#> $db.version
#> [1] "3.35.2"
#> 
#> $dbname
#> [1] "C:\\Users\\{username}\\AppData\\Local\\Temp\\Rtmpy4fGoA\\file401c136a1aa7"
#> 
#> $username
#> [1] NA
#> 
#> $host
#> [1] NA
#> 
#> $port
#> [1] NA
DBI::dbGetInfo(con_duckdb)
#> $dbname
#> character(0)
#> 
#> $db.version
#> [1] NA
#> 
#> $username
#> [1] NA
#> 
#> $host
#> [1] NA
#> 
#> $port
#> [1] NA
```

<sup>Created on 2021-04-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

### PR DESCRIPTION
**Pull request by hannes**
Fixes #1600

### CODE REVIEW

**Comment by hannesmuehleisen** [comment]
@ateucher I am getting strange errors in the CI on this one, `Error: cannot change value of locked binding for '.sys.timezone'`, could that be related to your PR #1302 ?

**Comment by hannesmuehleisen** [comment]
Indeed this also fails on R-devel for me now

**Comment by hannesmuehleisen** [comment]
This breaks because of a recent change in R: https://github.com/r-lib/withr/issues/165
Added hacky workaround with `unlockBinding(".sys.timezone", baseenv())`

File: tools/rpkg/tests/testthat/test_timezone.R [hunk 1]
```diff
@@ -26,6 +26,7 @@ test_that("timezone_out works with a specified timezone", {
 })
 
 test_that("timezone_out works with '' and converts to local timezime", {
+  unlockBinding(".sys.timezone", baseenv())
```
Comment (ateucher) [inline tools/rpkg/tests/testthat/test_timezone.R:29]:
```suggestion
  unlockBinding(".sys.timezone", baseenv())
  on.exit(lockBinding(".sys.timezone", baseenv()), add = TRUE)
```

File: tools/rpkg/tests/testthat/test_timezone.R [hunk 2]
```diff
@@ -36,6 +37,7 @@ test_that("timezone_out works with '' and converts to local timezime", {
 })
 
 test_that("timezone_out works with Sys.timezone", {
+  unlockBinding(".sys.timezone", baseenv())
```
Comment (ateucher) [inline tools/rpkg/tests/testthat/test_timezone.R:40]:
```suggestion
  unlockBinding(".sys.timezone", baseenv())
    on.exit(lockBinding(".sys.timezone", baseenv()), add = TRUE)
```

File: tools/rpkg/tests/testthat/test_timezone.R [hunk 3]
```diff
@@ -64,6 +66,7 @@ test_that("timezone_out works with a specified timezone and tz_out_convert = 'fo
 })
 
 test_that("timezone_out works with '' and tz_out_convert = 'force': forces local timezime", {
+  unlockBinding(".sys.timezone", baseenv())
```
Comment (ateucher) [inline tools/rpkg/tests/testthat/test_timezone.R:69]:
```suggestion
  unlockBinding(".sys.timezone", baseenv())
  on.exit(lockBinding(".sys.timezone", baseenv()), add = TRUE)
```

File: tools/rpkg/tests/testthat/test_timezone.R [hunk 4]
```diff
@@ -74,6 +77,7 @@ test_that("timezone_out works with '' and tz_out_convert = 'force': forces local
 })
 
 test_that("timezone_out works with a specified local timezone and tz_out_convert = 'force': forces local timezime", {
+  unlockBinding(".sys.timezone", baseenv())
```
Comment (ateucher) [inline tools/rpkg/tests/testthat/test_timezone.R:80]:
```suggestion
  unlockBinding(".sys.timezone", baseenv())
  on.exit(lockBinding(".sys.timezone", baseenv()), add = TRUE)
```

**Comment by ateucher** [comment]
Ok so it's just in the tests.... good to hear because Ily don't remember modifying `.sys.timezone` in that PR. It's probably a good idea to re-lock it when exiting the tests... I added the suggestions above.

**Pull request merged**